### PR TITLE
Added discharge patient filters

### PR DESCRIPTION
**Pull request by shivankacker**
## Proposed Changes

- Adds new advance filters for discharged patients.

### Associated Issue

- https://github.com/coronasafe/care_fe/issues/7737

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

File: care/facility/api/viewsets/patient.py [hunk 1]
```diff
@@ -602,7 +602,45 @@ def transfer(self, request, *args, **kwargs):
...
 
 class FacilityDischargedPatientFilterSet(filters.FilterSet):
+    disease_status = CareChoiceFilter(choice_dict=DISEASE_STATUS_DICT)
+    phone_number = filters.CharFilter(field_name="phone_number")
+    emergency_phone_number = filters.CharFilter(field_name="emergency_phone_number")
     name = filters.CharFilter(field_name="name", lookup_expr="icontains")
+    gender = filters.NumberFilter(field_name="gender")
+    age = filters.NumberFilter(field_name="age")
+    age_min = filters.NumberFilter(field_name="age", lookup_expr="gte")
+    age_max = filters.NumberFilter(field_name="age", lookup_expr="lte")
```
Comment (rithviknishad) [inline care/facility/api/viewsets/patient.py:613]: Does the `age` filters work? You'll need to annotate the QuerySet with age and in Discharged Patient Viewset's Queryset it is not annotated.

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by aparnacoronasafe** [comment]
@skks1212 Will this be ready for production in next release? next monday?

**Comment by shivankacker** [comment]
@aparnacoronasafe will try my best to get it reviewed and merged by then

**Review by rithviknishad — APPROVED** [review]

**Review by sainak — APPROVED** [review]

**Comment by vigneshhari** [comment]
@sainak create a ticket to abstract repeated code. We can merge this in for now.

**Comment by sainak** [comment]
> @sainak create a ticket to abstract repeated code. We can merge this in for now.

tracked in #2165

**Pull request merged**